.github/actions/checkBundleVersionStringMatch/checkBundleVersionStringMatch.js

```diff
@@ -0,0 +1,14 @@
+const core = require('@actions/core');
+const {execSync} = require('child_process');
+const {PLIST_PATH} = require('../../libs/nativeVersionUpdater');
+
+const bundleVersion = execSync(`/usr/libexec/PlistBuddy -c "Print :CFBundleVersion" ${PLIST_PATH}`);
+const shortBundleVersion = execSync(`/usr/libexec/PlistBuddy -c "Print :CFBundleShortVersionString" ${PLIST_PATH}`);
+
+if (shortBundleVersion !== (bundleVersion.split('-') || [''])[0]) {
```

Comment:
NAB: This comment is more for me to learn. I see in `nativeVersionUpdater.js` that the shortBundleVersion is obtained as following: https://github.com/Expensify/App/blob/fa2e8a9b172d14e51d92b7a916a357328655fca8/.github/libs/nativeVersionUpdater.js#L79

That means in terms of your example from the PR body: `1-1-1-0` this would be `1` and then on the next line we add another zero to the version:
https://github.com/Expensify/App/blob/fa2e8a9b172d14e51d92b7a916a357328655fca8/.github/libs/nativeVersionUpdater.js#L80

What I do not understand is taking the first - `[0]` element of the split would not check for the case you have described in the PR body:
- Compatible bundle version strings: `1.1.1` and `1.1.1.0`
- Incompatible bundle version strings: `1.1.1` and `1.1.0.0`

Thank you for explanation of this.

Reply:
Ah, no worries! So in this case we're grabbing the version number from the [info.plist](https://github.com/Expensify/App/blob/aafb09e0f45ea1480df0b934114ab18058b75d07/ios/NewExpensify/Info.plist#L34), which has the versions formatted as `x.x.x-x`. So if the bundleVersion is `1.1.1-0` and the shortBundleVersion is `1.1.1`, then they would match. Let me know if that makes sense!